File: src/main/java/voidchess/board/SimpleChessBoardInterface.java
package voidchess.board;

import voidchess.figures.Figure;
import voidchess.helper.CheckStatus;
import voidchess.helper.Position;

/**
 * @author stephan
 */
public interface SimpleChessBoardInterface extends BasicChessGameInterface {
    void init();

    void init(int chess960);

    void init(String des);

    void setFigure(Position pos, Figure figure);

    boolean isCheck(boolean isWhite);

    CheckStatus getCheckStatus(boolean isWhite);
}

File: src/main/kotlin/voidchess/board/SimpleChessBoardInterface.kt
package voidchess.board

import voidchess.board.check.AttackLines
import voidchess.figures.Figure
import voidchess.board.move.Position


interface SimpleChessBoardInterface : BasicChessGameInterface {

    fun init()
    fun init(chess960: Int)
    fun init(des: String)
    fun setFigure(pos: Position, figure: Figure)
    fun clearFigure(pos: Position)
    // moves figure, returns the figure that was taken
    // "normal" moves only, no special cases (castling, enpassant, pawn promotion)
    fun move(figure: Figure, to: Position): Figure?
    fun undoMove(figure: Figure, from: Position, figureTaken: Figure?)
    fun isCheck(isWhite: Boolean): Boolean
    fun getCachedAttackLines(isWhite: Boolean): AttackLines
}
